app/controller.go

```diff
@@ -157,6 +158,36 @@ func (c *Controller) CreateHost(w http.ResponseWriter, r *http.Request, user Use
 	return nil
 }
 
+func (c *Controller) ListHosts(w http.ResponseWriter, r *http.Request, user UserInfo) error {
+	listReq, err := BuildListHostRequest(r)
+	if err != nil {
+		return err
+	}
+	res, err := c.instanceManager.ListHosts(getZone(r), user, listReq)
+	if err != nil {
+		return err
+	}
+	replyJSON(w, res, http.StatusOK)
+	return nil
+}
+
+func BuildListHostRequest(r *http.Request) (*ListHostsRequest, error) {
+	var maxResults int
+	maxResultsRaw := r.URL.Query().Get("maxResults")
+	if maxResultsRaw != "" {
```

Comment:
might be worth it to encapsulate this in a utility function to parse int query parameters

Reply:
done